bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/ShellyHandlerFactory.java

```diff
@@ -0,0 +1,212 @@
+
+import static org.openhab.binding.shelly.internal.ShellyBindingConstants.*;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Dictionary;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+import org.apache.commons.lang.Validate;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.net.NetworkAddressService;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.openhab.binding.shelly.internal.coap.ShellyCoapServer;
+import org.openhab.binding.shelly.internal.config.ShellyBindingConfiguration;
+import org.openhab.binding.shelly.internal.handler.ShellyDeviceListener;
+import org.openhab.binding.shelly.internal.handler.ShellyLightHandler;
+import org.openhab.binding.shelly.internal.handler.ShellyRelayHandler;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.Version;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ShellyHandlerFactory} is responsible for creating things and thing handlers.
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = { ThingHandlerFactory.class,
+        ShellyHandlerFactory.class }, immediate = true, configurationPid = "binding.shelly")
+public class ShellyHandlerFactory extends BaseThingHandlerFactory {
+    private final Logger logger = LoggerFactory.getLogger(ShellyHandlerFactory.class);
+
+    @Nullable
+    private NetworkAddressService networkAddressService;
+    private final Set<ShellyDeviceListener> deviceListeners = new CopyOnWriteArraySet<>();
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = ShellyBindingConstants.SUPPORTED_THING_TYPES_UIDS;
+    private static boolean initialized = false;
+    private ShellyBindingConfiguration bindingConfig = new ShellyBindingConfiguration();
+    @Nullable
+    private ShellyCoapServer coapServer;
+
+    /**
+     * Activate the bundle: save properties
+     *
+     * @param componentContext
+     * @param configProperties set of properties from cfg (use same names as in
+     *            thing config)
+     */
+    @Activate
+    protected void activate(ComponentContext componentContext, Map<String, Object> configProperties) {
+        try {
+            super.activate(componentContext);
+
+            try {
+                if (!initialized) {
+                    Bundle bundle = this.getBundleContext().getBundle();
```

Comment:
What's the purpose of this, please elaborate

Reply:
What do you mean?
This allows to easily identify the installed version. Due to the fact that the user could get the jar from different sources (not just the dist package) I want to keep this info.

Reply:
This is a very uncommon usage. The installed binding version can be checked at karaf console using bundle:list command.

Reply:
All bindings being merged here should follow certain standards. This logging of the bundle version should not be done here. Because it's very generic and not specific for a binding. If such a feature is wanted this should not be implemented for each binding. So please remove this part.

Reply:
not happy with that. this was a request by the community and helped to validate that the new version is active (rather than cache problems). Yes, I could use the OH console and use bundles:list, but this approach makes it way easier.